ftl/python/builder.py

```diff
@@ -27,82 +27,109 @@
 
 _PYTHON_NAMESPACE = 'python-requirements-cache'
 _REQUIREMENTS_TXT = 'requirements.txt'
+_WORK_DIRS = ['app', 'env', 'wheel']
 
 
 class Python(builder.JustApp):
     def __init__(self, ctx):
         self.descriptor_files = [_REQUIREMENTS_TXT]
         self.namespace = _PYTHON_NAMESPACE
+        self.ctx = ctx
         super(Python, self).__init__(ctx)
 
     def __enter__(self):
         """Override."""
         return self
 
-    def _generate_overrides(self):
+    def _generate_overrides(self, py_version='python2.7'):
         return metadata.Overrides(
-            creation_time=str(datetime.date.today()) + "T00:00:00Z")
-
-    def CreatePackageBase(self, base_image, cache):
+            creation_time=str(datetime.date.today()) + "T00:00:00Z",
+            env={
+                "VIRTUAL_ENV": "/env",  # not working properly, not sure why
+                "PYTHONUSERBASE": "/env",  # this is adding the deps
+                "PATH": "/env/bin:$PATH"
+            })
+
+    def CreatePackageBase(self, base_image, py_version=2.7):
         """Override."""
-        overrides = self._generate_overrides()
-
-        layer, sha = self._gen_package_tar()
-        logging.info('Generated layer with sha: %s', sha)
-
-        with append.Layer(
-                base_image, layer, diff_id=sha,
-                overrides=overrides) as dep_image:
-            return dep_image
+        overrides = self._generate_overrides(py_version)
+
+        tmp_app, venv_dir, wheel_dir = self._gen_dirs(_WORK_DIRS)
+        whls = self._resolve_whls(tmp_app, venv_dir, wheel_dir)
+        pkg_dirs = [self._whl_to_fslayer(whl, venv_dir) for whl in whls]
+        logging.info("pkg_dirs" + str(pkg_dirs))
+        for pkg_dir in pkg_dirs:
+            layer, sha = self._gen_package_tar(pkg_dir)
+            logging.info('Generated layer with sha: %s', sha)
+            base_image = append.Layer(
```

Comment:
I see that you're iteratively adding layers to "base_image" and re-using the same variable name, but it might be more clear if you make a copy once at the start:

```python
package_base = base_image
```

Then add and re-use package_base.

Reply:
done